### PR TITLE
Refactor/crypto ciphers

### PR DESCRIPTION
**Pull request by magsteen**
Only had method for cipher generation of public key. Added one for secret keys too.

### CODE REVIEW

File: core/src/consumer.rs [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (tomrsae) [inline core/src/consumer.rs]: Remove this entire file from this PR to prevent unnecessary merge conflicts later on

File: core/src/lib.rs [hunk 2]
```diff
@@ -1 +1,2 @@
+pub mod consumer;
```
Comment (tomrsae) [inline core/src/lib.rs]: Remove from this PR to prevent unnecessary merge conflicts

File: core/src/crypto.rs [hunk 3]
```diff
@@ -25,8 +25,12 @@ impl Secret {
         self.secret.diffie_hellman(pub_key)
     }
 
-    pub fn gen_cipher(&mut self) -> Aes256 {
+    pub fn gen_pub_cipher(&mut self) -> Aes256 {
```
Comment (tomrsae) [inline core/src/crypto.rs]: Perhaps create a generic `gen_cipher` method that both these methods utilize? Just a thought, but makes for more modularized code and easier maintenance.

Comment (tomrsae) [inline core/src/crypto.rs]: Also no need for `mut` in either of the methods, as far as I can tell.

Comment (magsteen) [reply core/src/crypto.rs]: gen_pub_cipher() needs a mut self, but the other doesn't no :)

**Review by tomrsae — APPROVED** [review]
Looks perfect, ready for merge!

**Pull request merged**